modules/storages/spec/common/storages/peripherals/storage_interaction/create_folder_command_spec.rb

```diff
@@ -0,0 +1,197 @@
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require "spec_helper"
+require_module_spec_helper
+
+RSpec.describe "create folder command" do
+  using Storages::Peripherals::ServiceResultRefinements
+
+  shared_examples_for "basic command setup" do
+    it "is registered as commands.create_folder" do
+      expect(Storages::Peripherals::Registry
+               .resolve("#{storage.short_provider_type}.commands.create_folder")).to eq(described_class)
+    end
+
+    it "responds to #call with correct parameters" do
+      expect(described_class).to respond_to(:call)
+
+      method = described_class.method(:call)
+      expect(method.parameters).to contain_exactly(%i[keyreq storage],
+                                                   %i[keyreq auth_strategy],
+                                                   %i[keyreq folder_name],
+                                                   %i[keyreq parent_location])
+    end
+  end
+
+  shared_examples_for "successful folder creation" do
+    it "creates a folder" do
+      result = described_class.call(storage:, auth_strategy:, folder_name:, parent_location:)
+
+      result.match(
+        on_success: ->(response) do
+          expect(response).to be_a(Storages::StorageFile)
+          expect(response.name).to eq(folder_name)
+          expect(response.location).to eq(path)
+        end,
+        on_failure: ->(error) { fail "Expected success, got #{error}" }
+      )
+    ensure
+      delete_created_folder(storage, result.result)
+    end
+  end
+
+  shared_examples_for "parent not found" do
+    it "returns a failure" do
+      result = described_class.call(storage:, auth_strategy:, folder_name:, parent_location:)
+
+      result.match(
+        on_failure: ->(error) do
+          expect(error.code).to eq(:not_found)
+          expect(error.data.source).to eq(error_source)
+        end,
+        on_success: ->(result) { fail "Expected failure, got #{result}" }
+      )
```

Comment:
🟡 IMO this style offers little in terms of clarity.

The suggestion below feels more idiomatic and is shorter. =)

```suggestion
      expect(result).to be_failure
   
      error = result.error
      expect(error.code).to eq(:not_found)
      expect(error.data.source).to eq(error_source)
```